plugins/validate/__init__.py

```diff
@@ -0,0 +1,126 @@
+    to_list,
+)
+
+try:
+    import yaml
+
+    try:
+        from yaml import CSafeLoader as SafeLoader
+    except ImportError:
+        from yaml import SafeLoader
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+
+class ValidateBase(object):
+    """The base class for data validators
+    Provides a  _debug function to normalize debug output
+    """
+
+    def __init__(self, data, criteria, engine, plugin_vars=None, kwargs=None):
+        self._data = data
+        self._criteria = criteria
+        self._engine = engine
+        self._plugin_vars = plugin_vars if plugin_vars is not None else {}
+        self._result = {}
+        self._kwargs = kwargs if kwargs is not None else {}
+        self._sub_plugin_options = {}
+
+        cref = dict(zip(["corg", "cname", "plugin"], engine.split(".")))
+        validatorlib = "ansible_collections.{corg}.{cname}.plugins.validate.{plugin}".format(
+            **cref
+        )
+
+        validatordoc = getattr(import_module(validatorlib), "DOCUMENTATION")
+        if validatordoc:
+            self._set_sub_plugin_options(validatordoc)
+
+    def _set_sub_plugin_options(self, doc):
+        params = {}
+        argspec = deepcopy(doc)
+        argspec_obj = yaml.load(argspec, SafeLoader)
+        options = argspec_obj.get("options", {})
+
+        if not options:
+            return None
+
+        for option_name, option_value in iteritems(options):
+
+            option_var_name_list = option_value.get("vars", [])
+            option_env_name_list = option_value.get("env", [])
+
+            # check if plugin configuration option passed as kwargs
+            # valid for lookup, filter, test plugins or pass through
+            # variables if supported by the module.
+            if option_name in self._kwargs:
+                params[option_name] = self._kwargs[option_name]
+                continue
+
+            # check if plugin configuration option passed in task vars eg.
+            #  vars:
+            #  - name: ansible_validate_jsonschema_draft
+            #  - name: ansible_validate_jsonschema_draft_type
+            if option_var_name_list and (option_name not in params):
+                for var_name_entry in to_list(option_var_name_list):
+                    if not isinstance(var_name_entry, dict):
+                        raise AnsibleError(
+                            "invalid type '%s' for the value of '%s' option,"
+                            " should to be type dict"
+                            % (type(var_name_entry), var_name_entry)
+                        )
+                    var_name = var_name_entry.get("name")
+                    if var_name and var_name in self._plugin_vars:
+                        params[option_name] = self._plugin_vars[var_name]
+                        break
+
+            # check if plugin configuration option as passed as enviornment  eg.
+            # env:
+            # - name: ANSIBLE_VALIDATE_JSONSCHEMA_DRAFT
```

Comment:
I'm on the fence about supporting env vars, think it's needed?

Reply:
I don't see any harm in not supporting it, other plugins support setting environment variable.

Reply:
sounds good